executor/index_merge_reader.go

```diff
@@ -510,6 +535,7 @@ func (e *IndexMergeReaderExecutor) startPartialTableWorker(ctx context.Context,
 					if err = worker.tableReader.Close(); err != nil {
 						logutil.Logger(ctx).Error("close Select result failed:", zap.Error(err))
 					}
+					tableReaderClosed = true
```

Comment:
If worker.tableReader.Close() return a error, it will be called again in defer()?

Reply:
Will fix in https://github.com/pingcap/tidb/issues/41647